### PR TITLE
fix: Correctly handle computed object assignments

### PR DESCRIPTION
**Pull request by kamilogorek**
Fixes https://github.com/getsentry/js-source-scopes/issues/16

### CODE REVIEW

File: src/swc.rs [hunk 1]
```diff
@@ -354,6 +355,18 @@ fn infer_name_from_expr(mut expr: &ast::Expr) -> Option<ScopeName> {
                         .push_front(NameComponent::ident(ident.clone()));
                     scope_name.components.push_front(NameComponent::interp("."));
                 }
+
+                if let Some(computed_prop) = member.prop.as_computed() {
+                    if let Some((computed_name, need_delimiter)) =
```
Comment (kamilogorek) [inline src/swc.rs:360]: I'm not convinced about `need_delimiter` but that was good enough ¯\_(ツ)_/¯

Comment (Swatinem) [reply src/swc.rs:360]: you can just pass `&mut scope_name.components` into the fn, that would be a lot simpler I believe

File: src/swc.rs [hunk 2]
```diff
@@ -369,6 +382,24 @@ fn infer_name_from_expr(mut expr: &ast::Expr) -> Option<ScopeName> {
     }
 }
 
+fn computed_prop_name_to_component(prop: &ast::ComputedPropName) -> Option<(NameComponent, bool)> {
+    if let Some(lit) = prop.expr.as_lit() {
+        if let ast::Lit::Str(prop) = lit {
+            return Some((NameComponent::interp(prop.value.to_string()), true));
```
Comment (Swatinem) [inline src/swc.rs:388]: a string is not necessarily a valid identifier, you could have `"all kinds of craaaaazy stuff"` in there ;-)
better to use `interp(format!("[{:?}]", prop.value))`. I believe we can just do that for all kinds of literals, as there could also be other types I believe?

File: src/swc.rs [hunk 3]
```diff
@@ -369,6 +382,24 @@ fn infer_name_from_expr(mut expr: &ast::Expr) -> Option<ScopeName> {
...
+        }
+
+        if let ast::Lit::Num(prop) = lit {
+            return Some((NameComponent::interp(format!("[{}]", prop.value)), false));
+        }
+    }
+
+    if let Some(ident) = prop.expr.as_ident() {
+        return Some((NameComponent::interp(format!("[{}]", ident.sym)), false));
```
Comment (Swatinem) [inline src/swc.rs:397]: the identifier itself could be a valid token that we can remap, turning `[a]` into `[name]`. so we could use:
`interp("["), ident(ident), interp("]")`

**Review by Swatinem — APPROVED** [review]

**Pull request merged**